### DISCUSSION TITLE
Design main screen displaying App Preview Failed To Load

### DISCUSSION THREAD

**Discussion by rochgohu** [Budibase/budibase#1979] [Help]
hi everyone,

I just installed Budibase via Docker and I get stuck with the Design Preview main area not loading. There's this message "App Preview Failed To Load"

![image](https://user-images.githubusercontent.com/7847468/125337377-ed043f00-e31c-11eb-83c6-71ed4e8505f7.png)

I don't know if it's related - in the console I could see that there was a few issues coming from a JSON parse

`Uncaught SyntaxError: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at receiveMessage (VM111 about:srcdoc:51)`

Thanks for your help

**Comment by aptkingston** [comment]
Hi @Bongzy13. I replied to your comment on that issue. Basically try exporting and importing your app again. Or just create a new app if this was only a test app anyway.

**Comment by hannesfant** [comment]
I'm also experiencing the same issue with 2 completely fresh apps. This is a fresh install, but my old installation from a few weeks ago works perfectly fine.

**Reply by joebudi** [reply]
Click the blue update icon beside the preview icon.

**Reply by hannesfant** [reply]
This?
![image](https://user-images.githubusercontent.com/16209734/127753237-77143f97-6120-4320-8e90-cc94411c3371.png)
Also running it behind nginx if that helps.

**Reply by aptkingston** [reply]
Hey @hannesfant, can I ask what browser you're using? I can see where the error is coming from, but I can't understand how it could possibly throw an error, so there must be something quite unique about your environment/browser/extensions that is interfering with passing messages between the builder and the app preview.

**Reply by hannesfant** [reply]
> Hey @hannesfant, can I ask what browser you're using? I can see where the error is coming from, but I can't understand how it could possibly throw an error, so there must be something quite unique about your environment/browser/extensions that is interfering with passing messages between the builder and the app preview.

Using Chrome 92.0.4515.107 on Windows. However I tried disabling Dashlane and now it works like expected. So not entirely sure what about Dashlane is interfering with it, but something for sure is.
For reference, this is the exact error I'm seeing:
![image](https://user-images.githubusercontent.com/16209734/127770223-9c8c650c-32c6-41a7-be86-8082ef626310.png)

**Reply by aptkingston** [reply]
Interesting, it must be some security feature of the extension that's interfering with posting messages to iframes. The cause of the error is that the app preview (which is inside an iframe) is receiving garbage instead of the real message, and it's throwing that error when trying to parse the garbage. Thanks a lot for discovering that Dashlane was the cause. It may also be that Dashlane is sending its own custom message to any iframe windows for some reason, which crashes it before it gets a chance to read the real message. I'll push a fix so at least it doesn't die when it gets a bad message, and then hopefully that might allow it to load after Dashlane has finished doing its thing.

**Comment by easyrun32** [comment]
```cd packages/server ```
```yarn predocker ```

if something happens in the future with that script here's what it ran 
```copyfiles -f ../client/dist/budibase-client.js ../standard-components/manifest.json client```

Bascially it takes the client/dist/budibase-client.js and copies  it somewhere else
